intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -928,3 +936,303 @@ def validate_ip(value):
+sandbox_url = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('malware.hash.md5', 'md5hash'),
+        ('event_description.url', 'url'),
+        ('extra.', 'user_agent', validate_to_none),
+        ('extra.', 'host', validate_to_none),
+        ('extra.', 'method', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.type': 'malware',
+        'feed.code': 'shadowserver-sandboxurl',
+        'classification.identifier': 'sandboxurl',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Spam-URL
+spam_url = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+    ],
+    'optional_fields': [
+        ('event_description.url', 'url'),
+        ('source.reverse_dns', 'host'),
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('extra.', 'subject', validate_to_none),
+        ('extra.', 'src', validate_to_none),
+        ('extra.', 'src_asn', validate_to_none),
+        ('extra.', 'src_geo', validate_to_none),
+        ('extra.', 'src_region', validate_to_none),
+        ('extra.', 'src_city', validate_to_none),
+        ('extra.', 'sender', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.type': 'spam',
+        'feed.code': 'shadowserver-spamurl',
+        'classification.identifier': 'spamurl',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-Proxy
+open_proxy = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('source.reverse_dns', 'dns'),
+        #('source.reverse_dns', 'hostname'),  # ..this is an old column name
+        ('protocol.application', 'type'),
+        ('extra.', 'password', validate_to_none),
+        ('extra.', 'os_name', validate_to_none),
+        ('extra.', 'os_version', validate_to_none),
+        ('event_description.text', 'via'),
+    ],
+    'constant_fields': {
+        'classification.type': 'other',
+        'feed.code': 'shadowserver-openproxy',
+        'classification.identifier': 'openproxy',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Sinkhole-HTTP-Referer
+sinkhole_http_referer = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        #('source.ip', 'ip'),  # ..this is an old column name
+        ('source.ip', 'inet'),
```

Comment:
Haven't seen this report since 2014, got this field's name from https://www.shadowserver.org/wiki/pmwiki.php/Services/Sinkhole-HTTP-Referer

Maybe someone else can check it.